packages/flutter/lib/src/material/mergeable_material.dart

```diff
@@ -539,6 +543,23 @@ class _MergeableMaterialState extends State<MergeableMaterial> {
   }
 }
 
+class _MergeableMaterialSliceKey extends GlobalKey {
```

Comment:
This is different since it can only be equal to another `_MergeableMaterialSliceKey`. It's basically made for this use case only.